### PR TITLE
Processing: Add Check Geometry algorithm and test: Duplicate Nodes

### PR DESCRIPTION
**Pull request by Djedouas**
## Description

Following https://github.com/qgis/QGIS/pull/59637 to add the duplicated node geometry check from the geometry checker plugin in the processing toolbox.

### CODE REVIEW

**Comment by Djedouas** [comment]
Unrelated test error

**Comment by Djedouas** [comment]
Unrelated CI error

File: src/analysis/processing/qgsalgorithmcheckgeometryduplicatenodes.cpp [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+}
+
+QString QgsGeometryCheckDuplicateNodesAlgorithm::groupId() const
+{
+  return QStringLiteral( "checkgeometry" );
+}
+
+QString QgsGeometryCheckDuplicateNodesAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm check the nodes of line and polygon geometries.\n"
```
Comment (Djedouas) [inline src/analysis/processing/qgsalgorithmcheckgeometryduplicatenodes.cpp:55]:
```suggestion
  return QObject::tr( "This algorithm checks the nodes of line and polygon geometries.\n"
```

**Review by lbartoletti — APPROVED** [review]
